src/index.tsx

```diff
@@ -424,10 +434,43 @@ class Cropper extends React.Component<CropperProps, State> {
       this.props.rotation,
       this.props.restrictPosition
     )
-    this.props.onCropComplete &&
+  }
+
+  emitCropAreaChange = () => {
```

Comment:
@ValentinH I've started coding this solution following our `emitCropData(isCropComplete = true)` discussed function, but in the end, it felt more natural to use this specific `emitCropAreaChange` function, that calls the `debouncedEmitOnCropComplete` function.

This way I do not need to manually call:

```javascript
this.emitCropData(false);
this.debouncedEmitCropData();
```

Inside `componentDidUpdate`. This would also trigger `onCropAreaChange` twice every time if the crop is controlled from an outside source.

Let me know what you think.

Reply:
I like having 2 distinct `emitCropData()` and  `emitCropAreaChange()`.

However, I see a few small issues here:
- `debouncedEmitOnCropComplete` should not be called inside `emitCropAreaChange` as this is changing the existing behaviour. This new function should only call the new `onCropAreaChange`.
- I think we should debounce `onCropAreaChange` and not `onCropComplete`
- even by doing the 2 updates above, there is a small issue with the debounced callback when zooming with the mouse scroll. Indeed, the debounced call is called 300ms after the last non-debounced call, but with outdated values. I don't know if what I mean is clear but you can see the examples output you have added, the data changed 300ms after the last scroll and is not correct.

If this is fine for you, I would simply remove the debounce logic as there is not performance issues at the moment, let's not optimise imaginary issues. 🙂

Reply:
> If this is fine for you, I would simply remove the debounce logic as there is not performance issues at the moment, let's not optimise imaginary issues. :slightly_smiling_face:

I think it would be the best thing to do too :slightly_smiling_face: . This way I can only focus on solving the specific problem of not having a `onCropAreaChange` callback to receive when adjusting the crop through external buttons. I will avoid a lot of unnecessary complexity and dealing with those issues pointed above.